repo2docker/app.py

```diff
@@ -535,15 +531,78 @@ def _get_free_port(self):
         s.close()
         return port
 
+    def find_image(self):
+        # check if we already have an image for this content
+        client = docker.APIClient(version='auto', **kwargs_from_env())
+        for image in client.images():
+            if image['RepoTags'] is not None:
+                for tag in image['RepoTags']:
+                    if tag == self.output_image_spec + ":latest":
+                        return True
+        return False
+
+    def _build_image(self, checkout_path):
```

Comment:
I think this can be folded into the build function

Reply:
I liked the separation because `build()` does a bunch of things before&after building, and `_build_image()` just builds the image. :-/ It is an attempt to make the functions a bit shorter. Happy to move it.

Reply:
Private conversation with @yuvipanda just now: we put everything back in one method and he is happy with that.

Maybe look at splitting at a later point.